GetIndexerStatus/index.js

```diff
@@ -2,6 +2,14 @@ const azureSearchRequest = require('../lib/AzureSearchRequest');
 
 module.exports = async function getIndexerStatus(context) {
   const indexerName = context.bindings.indexerName;
-  const response = await azureSearchRequest(`indexers/${indexerName}/status`, 'get');
-  return response.body.lastResult.status;
+  try {
+    const response = await azureSearchRequest(`indexers/${indexerName}/status`, 'get');
+
+    if (response.body.status === 'error') {
```

Comment:
Discussed this offline and concluded that we should explicitly handle any future changes to the API rather than let then 'fall through' as would be the case if we use the negated check above.

To this end, the intention is to add checks as to which API Version Id's the code supports (currently 2017-11-11 and 2017-11-11-preview) whilst also allowing the API Version Id to be passed as a parameter by the consumer.